_posts/2022-01-17-changelog-163.markdown

```diff
@@ -0,0 +1,146 @@
+
+* [#8144] Detect [E0747](https://doc.rust-lang.org/error-index.html#E0747) compiler error
+
+* [#8289] Prioritize expressions that can be implicitly coerced to the expected type when suggesting completion variants
+
+* [#8335] Do not run non-syntax inspections on files disabled by `cfg` attributes
+
+* [#8323] Process available items generated by build script and expand available procedural macro calls even if the whole project host code (build scripts and procedural macro libraries) cannot be successfully compiled and executed.
+Note, build script evaluation is disabled by default for now. To turn it on, enable `org.rust.cargo.evaluate.build.scripts` [experimental](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) feature
+
+* [#8119] Enable new name resolution inside functions bodies. It allows resolving [macros 2.0](https://rust-lang.github.io/rfcs/1584-macros.html) declared inside function bodies
+
+## Fixes
+
+* Auto-import fixes:
+    * [#8291] Improve sorting of auto-import variants
+
+    * [#8288] Fix auto-import from `alloc` crate when using [`#![no_std]`](https://docs.rust-embedded.org/book/intro/no-std.html)
+
+    * [#8286] Don't show auto-import from auxiliary stdlib crates such as `unwind`
+
+* [#8362] Do not produce a false-positive error about missing format macro arguments without `format_args_capture` feature starting with 1.58
+
+* [#8334] Don't perform [project model reloading](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-cargo-tool-window.html#cargo-load) when another IDE project become [trusted](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-cargo-tool-window.html#trust-project)
+
+* [#8318] Expand proc macros from `dev-dependencies`. Note, procedural macro expansion is still under development, and it's disabled by default for now. To turn it on, enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental features](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
+
+* [#8297] Fix default parameters handling in `Wrong number of type arguments` inspection
+
+* [#8279] Fix resolve of imports inside detached files
+
+* [#8269] Correctly parenthesize block-like expressions (e.g. `match`) in [`Inline Variable` refactoring](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#extractvar-refactoring) (by [@Kobzol])
+
+* [#8268] Fix type inference if there are dependencies named the same as stdlib crates
+
+* [#8263] Infer generic parameter types in `Convert closure to function` intention (by [@Kobzol])
+
+* [#8230] [`Introduce local variable` refactoring](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#extractvar-refactoring) no longer moves the new binding into closure bodies if it is used outside of them (by [@afetisov])
```

Comment:
Note, currently, it's rendered in the following way
![image](https://user-images.githubusercontent.com/2539310/149757326-5cb7b1d7-54f0-4eca-ab79-d32151264518.png)

I can't come up with a better solution than
```suggestion
* [#8230]&nbsp;[`Introduce local variable` refactoring](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#extractvar-refactoring) no longer moves the new binding into closure bodies if it is used outside of them (by [@afetisov])
```